### PR TITLE
DAOS-8173 test: Use positional parameter for daos container command a…

### PR DESCRIPTION
**Pull request by shimizukko**
…nd fixed tests

Use CommandWithPositionalParameters in daos command.
Use PositionalParameter for self.pool since most of the daos container
subcommands have pool as positional parameter.
Reordered the parameters and the methods to match that of the help.

Renamed pool_list_cont to container_list and use JSON. Fixed the tests to
incorporate this change.

Skip-unit-test: true
Test-tag: pr dmg_pool_evict list_containers
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/container/list_containers.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def create_list(self, count, pool_uuid, expected_uuids):
                 self.daos_cmd.get_output("container_create", **kwargs)[0])
         expected_uuids.sort()
         # daos pool list-cont returns container UUIDs as a list.
-        data = self.daos_cmd.pool_list_cont(**kwargs)
-        actual_uuids = data["uuids"]
+        data = self.daos_cmd.container_list(**kwargs)
+        actual_uuids = [uuid_label["UUID"] for uuid_label in data["response"]]
```
Comment (phender) [inline src/tests/ftest/container/list_containers.py:47]: Until we start creating containers with labels should we be verifying the rest of the `daos container list` command, e.g. `data["Label"] == "container_label_not_set"`

File: src/tests/ftest/util/daos_utils_base.py [hunk 2]
```diff
@@ -212,11 +214,31 @@ def __init__(self, sub_command):
...
+
+            def __init__(self):
+                """Create a daos container check command object."""
+                super(
+                    DaosCommandBase.ContainerSubCommand.CheckSubCommand,
+                    self).__init__("check")
+                self.cont = PositionalParameter(2)
+                self.src = FormattedParameter("--epc={}")
+
+        class CloneSubCommand(CommonContainerSubCommand):
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:233]:
```suggestion
        class CloneSubCommand(CommandWithParameters):
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 3]
```diff
@@ -212,11 +214,31 @@ def __init__(self, sub_command):
                 super().__init__(
                         "/run/daos/container/{}/*".format(sub_command),
                         sub_command)
-                self.pool = FormattedParameter("--pool={}")
+                self.pool = PositionalParameter(1)
                 self.sys_name = FormattedParameter("--sys-name={}")
                 self.cont = FormattedParameter("--cont={}")
                 self.path = FormattedParameter("--path={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:220]:
```suggestion
                self.sys_name = FormattedParameter("--sys-name={}")
                self.path = FormattedParameter("--path={}")
                self.pool = PositionalParameter(1)
                self.cont = PositionalParameter(2)
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 4]
```diff
@@ -212,11 +214,31 @@ def __init__(self, sub_command):
...
 
+        class CheckSubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container check command."""
+
+            def __init__(self):
+                """Create a daos container check command object."""
+                super(
+                    DaosCommandBase.ContainerSubCommand.CheckSubCommand,
+                    self).__init__("check")
+                self.cont = PositionalParameter(2)
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:230]:
```suggestion
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 5]
```diff
@@ -253,45 +275,49 @@ def __init__(self):
                 #           input file containing ACL
                 self.acl_file = FormattedParameter("--acl-file={}", None)
 
-        class CloneSubCommand(CommonContainerSubCommand):
-            """Defines an object for the daos container clone command."""
+        class CreateSnapSubCommand(CommonContainerSubCommand):
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:278]:
```suggestion

                #    -c, --cont=<UUID>
                #            container UUID (optional)
                self.cont = FormattedParameter("--cont={}")

        class CreateSnapSubCommand(CommonContainerSubCommand):
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 6]
```diff
@@ -253,45 +275,49 @@ def __init__(self):
...
-            """Defines an object for the daos container destroy command."""
+        class DelAttrSubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container del-attrs command."""
 
             def __init__(self):
-                """Create a daos container destroy command object."""
-                super().__init__("destroy")
-                self.force = FormattedParameter("--force", False)
+                """Create a daos container del-attrs command object."""
+                super().__init__("del-attrs")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:292]:
```suggestion
                super().__init__("del-attr")
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 7]
```diff
@@ -253,45 +275,49 @@ def __init__(self):
...
+                self.attr = FormattedParameter("--attr={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:293]:
```suggestion
                self.attr = PositionalParameter(3)
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 8]
```diff
@@ -253,45 +275,49 @@ def __init__(self):
...
-            """Defines an object for the daos container query command."""
+        class DestroySubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container destroy command."""
 
             def __init__(self):
-                """Create a daos container query command object."""
-                super().__init__("query")
+                """Create a daos container destroy command object."""
+                super().__init__("destroy")
+                self.cont = PositionalParameter(2)
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:309]:
```suggestion
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 9]
```diff
@@ -253,45 +275,49 @@ def __init__(self):
...
+            """Defines an object for the daos container destroy-snap command."""
 
             def __init__(self):
-                """Create a daos container check command object."""
-                super(
-                    DaosCommandBase.ContainerSubCommand.CheckSubCommand,
-                    self).__init__("check")
+                """Create a daos container destroy-snap command object."""
+                super().__init__("destroy-snap")
+                self.snap = FormattedParameter("--snap={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:318]:
```suggestion
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 10]
```diff
@@ -307,37 +333,29 @@ def __init__(self):
...
-            """Defines an object for the daos container overwrite-acl cmd."""
-
-            def __init__(self):
-                """Create a daos container overwrite-acl command object."""
-                super().__init__("overwrite-acl")
-                self.acl_file = FormattedParameter("--acl-file={}")
-
-        class UpdateAclSubCommand(CommonContainerSubCommand):
-            """Defines an object for the daos container update-acl command."""
+        class GetAttrSubCommand(CommonContainerSubCommand):
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:336]:
```suggestion
                #     -f, --force
                #             overwrite existing outfile
                self.force = FormattedParameter("--force", False)

        class GetAttrSubCommand(CommonContainerSubCommand):
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 11]
```diff
@@ -307,37 +333,29 @@ def __init__(self):
...
+            """Defines an object for the daos container get-attr command."""
 
             def __init__(self):
-                """Create a daos container update-acl command object."""
-                super().__init__("update-acl")
-                self.acl_file = FormattedParameter("--acl-file={}")
-                self.entry = FormattedParameter("--entry={}")
+                """Create a daos container get-attr command object."""
+                super().__init__("get-attr")
+                self.attr = FormattedParameter("--attr={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:342]:
```suggestion
                self.attr = PositionalParameter(3)
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 12]
```diff
@@ -307,37 +333,29 @@ def __init__(self):
...
+        class GetPropSubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container get-prop command."""
 
             def __init__(self):
-                """Create a daos container delete-acl command object."""
-                super().__init__("delete-acl")
-                self.principal = FormattedParameter("--principal={}")
+                """Create a daos container get-prop command object."""
+                super().__init__("get-prop")
+                self.prop = FormattedParameter("--prop={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:350]:
```suggestion
                self.prop = FormattedParameter("--properties={}")
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 13]
```diff
@@ -307,37 +333,29 @@ def __init__(self):
...
-            """Defines an object for the daos container stat command."""
+        class ListSubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container list command."""
 
             def __init__(self):
-                """Create a daos container stat command object."""
-                super().__init__("stat")
+                """Create a daos container list command object."""
+                super().__init__("list")
+                self.sys_name = FormattedParameter("--sys-name={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:358]:
```suggestion
                self.sys_name = FormattedParameter("--sys-name={}")
                self.pool = PositionalParameter(1)
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 14]
```diff
@@ -307,37 +333,29 @@ def __init__(self):
...
-                """Create a daos container delete-acl command object."""
-                super().__init__("delete-acl")
-                self.principal = FormattedParameter("--principal={}")
+                """Create a daos container get-prop command object."""
+                super().__init__("get-prop")
+                self.prop = FormattedParameter("--prop={}")
 
-        class StatSubCommand(CommonContainerSubCommand):
-            """Defines an object for the daos container stat command."""
+        class ListSubCommand(CommonContainerSubCommand):
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:352]:
```suggestion
        class ListSubCommand(CommandWithPositionalParameters):
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 15]
```diff
@@ -346,46 +364,52 @@ def __init__(self):
                 """Create a daos container list-attrs command object."""
                 super().__init__("list-attrs")
 
-        class DelAttrSubCommand(CommonContainerSubCommand):
-            """Defines an object for the daos container del-attrs command."""
+        class ListObjectsSubCommand(CommonContainerSubCommand):
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:367]:
```suggestion
                self.verbose = FormattedParameter("--verbose", False)

        class ListObjectsSubCommand(CommonContainerSubCommand):
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 16]
```diff
@@ -346,46 +364,52 @@ def __init__(self):
...
+            """Defines an object for the daos container list-objects command."""
 
             def __init__(self):
-                """Create a daos container del-attrs command object."""
-                super().__init__("del-attrs")
-                self.attr = FormattedParameter("--attr={}")
+                """Create a daos container list-objects command object."""
+                super().__init__("list-objects")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:372]:
```suggestion
                super().__init__("list-objects")
                self.epc = FormattedParameter("--epc={}")
```

File: src/tests/ftest/util/daos_utils_base.py [hunk 17]
```diff
@@ -346,46 +364,52 @@ def __init__(self):
...
+                self.epc = FormattedParameter("--epc={}")
+
+        class SetAttrSubCommand(CommonContainerSubCommand):
+            """Defines an object for the daos container set-attr command."""
+
+            def __init__(self):
+                """Create a daos container set-attr command object."""
+                super().__init__("set-attr")
+                self.attr = FormattedParameter("--attr={}")
+                self.value = FormattedParameter("--value={}")
```
Comment (phender) [inline src/tests/ftest/util/daos_utils_base.py:412]:
```suggestion
                self.attr = PositionalParameter(3)
                self.value = PositionalParameter(4)
```

**Review by phender — CHANGES_REQUESTED** [review]

**Review by phender — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]
LGTM.

**Comment by shimizukko** [comment]
Known failures that aren't relevant to my change:
* ior/ior_small.py - DAOS-8233
* pool/create_capacity_test.py - DAOS-8229

**Pull request merged**